packages/decoupled/src/cache/cache.ts

```diff
@@ -52,4 +52,8 @@ export class Cache extends SiteDependent implements CacheInterface {
         this.cache = (redis) ? new RedisCache(this.site) : new NodeCache(this.site);
 
     }
+
+    public async fetch(apiFetch: CallableFunction, { type, params }) {
```

Comment:
I think Cache.fetch should be more generic. It's more like a regular HTTP-fetch or request for which the result is cached (either on a custom ttl or the response's max-age). This fetch can be used by other components to actually fetch stuff, so it needs the same options like the original node-fetch. Instead of providing type, it will receive a full URL.